parser/src/Parse/Module.hs

```diff
@@ -96,27 +96,29 @@ import' =
           (_, postAs) <- whitespace
           (,) preAs <$> (,) postAs <$> capVar <?> ("an alias for module `" ++ moduleName ++ "`")
 
-    exposing :: IParser (Comments, PreCommented (Var.Listing Var.Value))
+    exposing :: IParser (Comments, PreCommented (Var.Listing Var.Value), Bool)
     exposing =
       do  (_, preExposing) <- try (whitespace <* reserved "exposing")
           (_, postExposing) <- whitespace
-          (postExposing2, listing') <- listing value
-          return (preExposing, (postExposing ++ postExposing2, listing'))
+          (postExposing2, listing', multiline) <- listing value
+          return (preExposing, (postExposing ++ postExposing2, listing'), multiline)
 
 
-listing :: IParser a -> IParser (Comments, Var.Listing a)
+listing :: IParser a -> IParser (Comments, Var.Listing a, Bool)
 listing item =
   expecting "a listing of values and types to expose, like (..)" $
   do  (_, preParen) <- try (whitespace <* char '(')
       (_, pre) <- whitespace
+      pushNewlineContext
       listing <-
           choice
-            [ (\_ pre post -> Var.OpenListing (Commented pre () post)) <$> string ".."
-            , (\x pre post -> Var.ExplicitListing (x pre post)) <$> commaSep1' item
+            [ (\_ pre post -> (Var.OpenListing (Commented pre () post))) <$> string ".."
+            , (\x pre post -> (Var.ExplicitListing (x pre post))) <$> commaSep1' item
             ]
+      sawNewline <- popNewlineContext
       (_, post) <- whitespace
```

Comment:
Same here, popNewlineContext should move down one line.  I will make this change as well.